src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -131,6 +131,7 @@ internal void SetWellKnownType(WellKnownType wellKnownType)
         private TypeFlags InitializeTypeFlags(TypeFlags mask)
         {
             TypeFlags flags = ComputeTypeFlags(mask);
+            flags = Context.ComputeTypeFlags(this, flags, mask);
```

Comment:
Making such a change requires that any mask specified to GetTypeFlags which is handled by the TypeSystemContext.ComputeTypeFlags must only incorporate a single set of flags, but it isn't unreasonable, and it would save a non-zero amount of excess work. (It helps that a single mask set is the only technique used in the typesystem today).